yt/loaders.py

```diff
@@ -1561,3 +1561,107 @@ def del_callback(self):
+        specify it here.  Otherwise, the fields are assumed to be at the root
+        level of the HDF5 file hierarchy.
+
+    fields : list of str, optional
+        The fields to be included as part of the dataset.  If this is not
+        included, all of the datasets under *root_node* will be included.
+        If your file contains, for instance, a "parameters" node at the root
+        level next to other fields, it would be (mistakenly) included.  This
+        allows you to specify only those that should be included.
+
+    bbox : array_like (xdim:zdim, LE:RE), optional
+        If supplied, this will be the bounding box for the dataset.  If not
+        supplied, it will be assumed to be from 0 to 1 in all dimensions.
+
+    nchunks : int, optional
+        How many chunks should this dataset be split into?  If 0 or not
+        supplied, yt will attempt to ensure that there is one chunk for every
+        64**3 zones in the dataset.
+
+    dataset_arguments : dict, optional
+        Any additional arguments that should be passed to
+        :class:`yt.loaders.load_amr_grids`, including things like the unit
+        length and the coordinates.
+
+    Returns
+    -------
+    :class:`yt.data_objects.static_output.Dataset` object
+        This returns an instance of a dataset, created with `load_amr_grids`,
+        that can read from the HDF5 file supplied to this function.  An open
+        handle to the HDF5 file is retained.
+
+    Raises
+    ------
+    FileNotFoundError
+        If fn does not match any existing file or directory.
+
+    """
+
+    from yt.utilities.on_demand_imports import _h5py as h5py
+
+    dataset_arguments = dataset_arguments or {}
+
+    if bbox is None:
+        bbox = np.array([[0.0, 1.0], [0.0, 1.0], [0.0, 1.0]])
+        mylog.info("Assuming unitary (0..1) bounding box.")
+
+    def _read_data(handle, root_node):
+        def _reader(grid, field_name):
+            ftype, fname = field_name
+            si = grid.get_global_startindex()
+            ei = si + grid.ActiveDimensions
+            return handle[root_node][fname][si[0] : ei[0], si[1] : ei[1], si[2] : ei[2]]
+
+        return _reader
+
+    handle = h5py.File(fn, "r")
+    reader = _read_data(handle, root_node)
+    if fields is None:
+        fields = list(handle[root_node].keys())
+        mylog.debug("Identified fields %s", fields)
+    shape = handle[root_node][fields[0]].shape
+    if nchunks <= 0:
+        # We apply a pretty simple heuristic here.  We don't want more than
+        # about 64^3 zones per chunk.  So ...
+        full_size = np.prod(shape)
+        nchunks = full_size // (64**3)
+        mylog.info("Auto-guessing %s chunks from a size of %s", nchunks, full_size)
+    grid_data = []
+    psize = get_psize(np.array(shape), nchunks)
+    left_edges, right_edges, shapes, _ = decompose_array(shape, psize, bbox)
+    for le, re, s in zip(left_edges, right_edges, shapes):
+        data = {_: reader for _ in fields}
```

Comment:
this is dependent on on my question/comment in `process_data` on including units... but if `process_data` were modified to allow `(callable, units)` tuples and if there was a way of getting units info passed in to this function, you could modify this line to pass that unit info along. For passing in units info -- we could allow the `fields` argument here to be a list of `(fieldname string, units string)` tuples?